daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SExpr.scala

```diff
@@ -339,13 +340,57 @@ object SExpr {
 
   /** The SEImportValue form is never constructed when compiling user LF.
     * It is only constructed at runtime by certain builtin-ops.
+    * Assumes the package needed to type `value` are preloaded.
     */
   final case class SEImportValue(typ: Ast.Type, value: V) extends SExpr {
     def execute(machine: Machine): Unit = {
       machine.importValue(typ, value)
     }
   }
 
+  /** Similarly to SEImportValue the SEImportValue form is never
+    * constructed when compiling user LF. It is only constructed
+    * at runtime by certain builtin-ops.
+    * Unlike SEImportValue, may throw a SpeedyHungry exception to
+    * request the packages of tyCon.
+    */
+  final case class SEImportInterface(
+      ifaceId: TypeConName,
+      coid: V.ContractId,
+      tmplId: TypeConName,
+      value: V,
+      alreadyRunOnce: Boolean = false,
+  ) extends SExpr {
+    def execute(machine: Machine): Unit = {
+      val pkgId = tmplId.packageId
+      if (!machine.compiledPackages.packageIds.contains(pkgId)) {
+        // We guard against loop in case there is a but somewhere else.
```

Comment:
I don’t understand what you are trying to guard against here. The sentence seems incomplete.

Reply:
as discussed s/but/bug/ I think it would be good to expand on the bug and expand the error message.